gzipi/lib.py

```diff
@@ -146,41 +169,139 @@
 _BYTES_IN_KiB = 1024
 
 
-def _is_valid_gzip_header(gzip_header):
+class Compression(enum.Enum):
+    NONE = None
+    GZIP = 'gzip'
+    ZSTD = 'zstd'
+
+
+SUPPORTED_COMPRESSIONS = {Compression.GZIP.value, Compression.ZSTD.value, }
+
+
+def get_compression_format(path: str) -> Optional[str]:
+    """Get compression format based on the extension."""
+    if not isinstance(path, str):
+        return Compression.NONE.value
+    if path.endswith('.gz'):
+        return Compression.GZIP.value
+    elif path.endswith('.zst'):
+        return Compression.ZSTD.value
+    else:
+        return Compression.NONE.value
+
+
+def _get_compression_type(path: str) -> Optional[str]:
+    with open(path, 'rb') as fin:
+        return _get_compression_type_from_fin(fin)
+
+
+def _get_compression_type_from_fin(fin: IO) -> Optional[str]:
+    header = fin.read(4)
```

Comment:
No need to read and seek, just do fin.peek(4).

Reply:
I don't think `peek` is supported for every IO. Even the `IO` definition from the typing module does not have such a signature.

Reply:
OK, fair point.